src/test/java/seedu/dengue/logic/commands/ClearCommandTest.java

```diff
@@ -68,8 +68,8 @@ public void execute_filteredNonEmptyDengueHotspotTracker_success() {
         Optional<Age> emptyAge = Optional.empty();
         Optional<Date> emptyDate = Optional.empty();
         Set<Variant> emptyVariants = new HashSet<>();
-        Range<Date> emptyDateRange = new Range<Date>(new StartDate(emptyDate), new EndDate(emptyDate));
-        Range<Age> emptyAgeRange = new Range<Age>(new StartAge(emptyAge), new EndAge(emptyAge));
+        Range<Date> emptyDateRange = new Range<>(new StartDate(emptyDate), new EndDate(emptyDate));
+        Range<Age> emptyAgeRange = new Range<>(new StartAge(emptyAge), new EndAge(emptyAge));
```

Comment:
```suggestion
        Range<Date> emptyDateRange = ContinuousData.generateRange(new StartDate(emptyDate), new EndDate(emptyDate));
        Range<Age> emptyAgeRange = ContinuousData.generateRange(new StartAge(emptyAge), new EndAge(emptyAge));
```

Help implement this: Abstract out range creation to make it safer (more defensive). I will make changes to ensure that the constructor for range can only be accessed via the ContinuousData interface

Reply:
Gotcha! Cool defensive coding.